### DISCUSSION TITLE
POST route not found

### DISCUSSION THREAD

**Discussion by hansajith1998** [livewire/livewire#5464] [7. Bugs (archived)]
After running composer update livewire upgraded to v2.10.8 and Suddenly Livewire requests are not found.
![image](https://user-images.githubusercontent.com/64520856/209468465-0a391cdf-b3fd-4c7d-8073-e091342e991c.png)

![image](https://user-images.githubusercontent.com/64520856/209468476-8db5d4a6-3b05-421a-9bfa-ca1975f055bb.png)

Then I have downgraded to v2.10.7. v2.10.7 version is working

**Comment by mbagusrd** [comment]
I think I got same problem here

**Comment by joshhanley** [comment]
@hansajith1998 and @mbagusrd there might be a bug in the new release. 

Any chance someone could create a small example repo which reproduces the issue with as little code as possible so we can investigate it?

Thanks!

**Reply by hansajith1998** [reply]
Fresh Laravel project with Jetstream package can easily reproduce the error
Jetstream livewire routes are gives `404`

> Repo 

`https://github.com/hansajith1998/livewire-v2.10.8-route-404-bug`

> Steps

```
composer create-project laravel/laravel livewire-v2.10.8-route-404-bug
cd livewire-v2.10.8-route-404-bug
composer require laravel/jetstream
php artisan jetstream:install livewire --teams
change env APP_URL ="http://localhost/livewire-v2.10.8-route-404-bug/public"
npm install
npm run build
php artisan migrate
```

> Need to run via "http://localhost/livewire-v2.10.8-route-404-bug/public". `php artisan serve` may not return this error.

**Reply by joshhanley** [reply]
@hansajith1998 thanks for that! 

Can you explain to me how you are running it under localhost (using Apache or Nginx)? What operating system are you using? Are you using anything else to get it running?

The reason I ask is because I use a Mac with Laravel Valet, so it uses Nginx and .test domains, as such want to make sure I can replicate it.

I’m also just curious as to why you have that configuration as Laravel strongly recommends not hosting an application from the root directory (so you should not have /public in your url).

Last question is, is this just breaking locally or is it breaking on your hosting server too?

Thanks!

**Reply by hansajith1998** [reply]
Apache server is running localhost

Its just a directs all requests to public/index.php file without running artisan. /public in url is only using locally. when production there are no /public in url. 


![image](https://user-images.githubusercontent.com/64520856/209599549-a5d0b3f0-04a8-4c71-a5ec-7e8ca870fd0d.png)

**Reply by hrvoj3e** [reply]
Filament PHP is broken right now when updated
 - `Upgrading livewire/livewire (v2.10.7 => v2.10.8)`

404 all around on every POST request

**Reply by Eighke** [reply]
I got the exact problem, after debugging I found out it because Laravel is on a non-root domain. It was "working" on version <= 2.10.7 because it was hidden by another bug (and it was fixed on 2.10.8 https://github.com/livewire/livewire/pull/4570)

Now we basically have the same issue as https://github.com/livewire/livewire/issues/1535 but on `\Livewire\Controllers\HttpConnectionHandler::applyPersistentMiddleware`

```php
$url = Livewire::originalUrl(); // Will return an absolute URL (e.g. https://localhost/myapp/some-route)

$request = Request::create($url, $method); // Will create a $request from it

$route = app('router')->getRoutes()->match($request); // Will fail, because /myapp/some-route doesn't match, but /some-route
```

It basically missing the basePath from the $request. 

I made a quick change to check and it's working

```
$request = Request::create(uri: $url, method: $method, server: ['SCRIPT_FILENAME' => '/myapp/index.php', 'PHP_SELF' => '/myapp/index.php']);
```

**Reply by ketankulkarni1** [reply]
> Filament PHP is broken right now when updated
> 
> * `Upgrading livewire/livewire (v2.10.7 => v2.10.8)`
> 
> 404 all around on every POST request

Hello, refer below mail by itzFinn,

I don't share their sentiment, but for some reason, I am getting notifications for livewire, starting from this exact comment. 

I wonder why I am getting these. I never forked this or even never even remotely contributed.

**Comment by joshhanley** [comment]
There was a bug released in v2.10.8. I have submitted a PR #5490 with a fix.

Thanks to @Eighke for the fix! 😁

Could someone please test out the PR and let me know if there are any issues? Thanks!

Hope this helps!

**Reply by Eighke** [reply]
Works well with the PR. :)

**Reply by farmaworld** [reply]
don't work for me

**Comment by itzfinn** [comment]
Good evening or morning. First, who is this emailing this to me when I am not a part of repository project. Second, how are you able to access this email account without having discreet permission of me.Please respond ASAP or I will report this to local authorities. Have a nice day.P.S.: conversation(s) of this thread will be record so don’t try anything snarky or stupid ok ? Sent from my iPhoneOn Jan 6, 2023, at 8:14 AM, Raffaele Filomena ***@***.***> wrote:﻿
don't work for me

**Reply by joshhanley** [reply]
@itzfinn I suspect that Github has had a bunch of bots leave spam comments on a heap of repositories that tagged random accounts, such as yours, that are unrelated to the project. I had a whole stack of unrelated notifications as well.

The issue is, once tagged, Github automatically subscribes people to the thread they are tagged in, even if all the bot comments have been removed. Hence why I believe you received an email regarding a new comment from Raffaele above saying "don't work for me".

I suspect it was Github's notification system that sent the email to your email address.

Hope this helps!

**Comment by orgabor** [comment]


**Reply by joshhanley** [reply]
@orgabor please comment on the PR not here so it can be worked through properly. 

Also if the PR doesn’t solve the issue, can you please provide steps for us to replicate it so we can ensure that it is fixed to include all cases. 

Thanks!

**Reply by orgabor** [reply]
@joshhanley ok, sure, I will attach it to the PR. I will also send you the steps
